src/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -76,12 +76,18 @@ private JObject InvokeFunction(JArray _params)
             UInt160 script_hash = UInt160.Parse(_params[0].AsString());
             string operation = _params[1].AsString();
             ContractParameter[] args = _params.Count >= 3 ? ((JArray)_params[2]).Select(p => ContractParameter.FromJson(p)).ToArray() : new ContractParameter[0];
+            CheckWitnessHashes checkWitnessHashes = null;
+            if (_params.Count > 3)
+            {
+                UInt160[] scriptHashesForVerifying = _params.Skip(3).Select(u => UInt160.Parse(u.AsString())).ToArray();
+                checkWitnessHashes = new CheckWitnessHashes(scriptHashesForVerifying);
+            }
```

Comment:
The new parameter can be an array.

Reply:
Have modified correspondingly:)

Reply:
I prefer accept only array, it's easier to understand.

Reply:
Array only please. Otherwise it is impossible to add new parameters in the future.

Reply:
Now only JArray format is reserved.